crates/rlp/src/header.rs

```diff
@@ -108,6 +108,40 @@ impl Header {
         core::str::from_utf8(bytes).map_err(|_| Error::Custom("invalid string"))
     }
 
+    /// Extracts the next payload from the given buffer, advancing it.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the buffer is too short, the header is invalid or one of the headers one
+    /// level deeper is invalid.
+    #[inline]
+    pub fn decode_raw<'a>(buf: &mut &'a [u8]) -> Result<PayloadView<'a>> {
+        let Self { list, payload_length } = Self::decode(buf)?;
+        let mut payload = &buf[..payload_length];
```

Comment:
Here and a few other places, this could be replaced with unsafe methods similar to `decode_bytes()`. However, since we'll be doing some `Vec` resizing anyway, I'm not sure it's worth the added complexity.

Reply:
Can we not use `decode_bytes` directly here?

The logic from `decode_bytes` could also be extracted to be separated from the `advance` call

Reply:
My first thought was to implement a private `decode` that returns the payload bytes as well as an `is_list`. 
However, if the `decode_bytes` then verifies this `is_list` afterwards, this would strictly speaking lead to a breaking change, as the behavior changes if the input is of an invalid type.

Technically, the `items.push(&payload[..rlp_length])` could also be rewritten with similar unsafe logic.

I am still thinking about the best way to consolidate the unsafe `advance` logic...

Reply:
I guess it would be nice, if we could use `split_at_unchecked()`

Reply:
Moved the following into a function that is used by both methods:
```rust
  if buf.remaining() < cnt {
      unreachable_unchecked()
  }
  let bytes = &buf[..cnt];
  buf.advance(cnt);
 ```